app/src/main/java/org/p2p/wallet/striga/model/StrigaDataLayerError.kt

```diff
@@ -10,8 +10,11 @@ sealed class StrigaDataLayerError(override val message: String) : Throwable() {
     ) : StrigaDataLayerError("Striga API unavailable, code: ${cause.code()} body: $body")
 
     class ApiServiceError(
-        val response: StrigaApiErrorResponse
-    ) : StrigaDataLayerError("Striga API returned error: code=${response.errorCode?.code} details=${response.details}")
+        val response: StrigaApiErrorResponse,
+    ) : StrigaDataLayerError("Striga API returned error: code=${response.errorCode.code} details=${response.details}") {
+        val errorCode: StrigaApiErrorCode
+            get() = response.errorCode
```

Comment:
we don't need `get` here since the response cannot be changed.
just make it ` val errorCode: StrigaApiErrorCode = response.errorCode`, so this value could return the cached value